src/main/java/appeng/block/networking/BlockWirelessConnector.java

```diff
@@ -0,0 +1,184 @@
+package appeng.block.networking;
+
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IIcon;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.ForgeDirection;
+
+import appeng.api.util.AEColor;
+import appeng.api.util.IOrientable;
+import appeng.block.AEBaseTileBlock;
+import appeng.client.texture.WirelessTextures;
+import appeng.core.features.AEFeature;
+import appeng.helpers.NullRotation;
+import appeng.tile.networking.TileWirelessConnector;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+
+public class BlockWirelessConnector extends AEBaseTileBlock {
+
+    public BlockWirelessConnector() {
+        super(Material.iron);
+        this.setTileEntity(TileWirelessConnector.class);
+        this.setFeature(EnumSet.of(AEFeature.Channels));
+    }
+
+    @Override
+    public void setHasSubtypes(boolean b) {
+        super.setHasSubtypes(true);
+    }
+
+    @Override
+    public ArrayList<ItemStack> getDrops(World world, int x, int y, int z, int metadata, int fortune) {
+        ItemStack is = new ItemStack(this);
+        TileWirelessConnector te = (TileWirelessConnector) world.getTileEntity(x, y, z);
+        if (te != null) {
+            if (te.getColor() != AEColor.Transparent) {
+                is.setItemDamage(te.getColor().ordinal() + 1);
+            }
+        }
+        ArrayList<ItemStack> arr = new ArrayList<>();
+        arr.add(is);
+        return arr;
+    }
+
+    @Override
+    public void addInformation(ItemStack is, EntityPlayer p, List<String> lines, boolean advancedItemTooltips) {
+        if (is.getItemDamage() == 0) {
+            lines.add(AEColor.Transparent.toString());
+        } else {
+            lines.add(AEColor.values()[is.getItemDamage() - 1].toString());
+        }
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public void getCheckedSubBlocks(Item item, CreativeTabs tabs, List<ItemStack> itemStacks) {
+        for (int i = 0; i < 17; i++) {
+            itemStacks.add(new ItemStack(this, 1, i));
+        }
+    }
+
+    @Override
+    public ItemStack getPickBlock(MovingObjectPosition target, World world, int x, int y, int z, EntityPlayer player) {
+        ItemStack is = new ItemStack(this);
+        TileWirelessConnector te = (TileWirelessConnector) world.getTileEntity(x, y, z);
+        if (te != null) {
```

Comment:
Instead of getting the TE and casting, then checking nonnull, you can get the TE without casting, then check instanceof. I.e.:
```java
TileEntity te = world.getTileEntity(x, y, z);
if (te instanceof TileWirelessConnector connector) {
    // ...
}
```
This way we avoid the *potential* bad cast, even if it is likely impossible to occur